ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/views/imagebrowser/repositories/AbstractRepository.java

```diff
@@ -142,7 +149,9 @@ protected void searchJarFile(final File jarFile, final IProgressMonitor monitor)
 			while ((entries.hasMoreElements()) && (!monitor.isCanceled())) {
 				ZipEntry entry = entries.nextElement();
 				if (isImageName(entry.getName())) {
-					addImageElement(new ImageElement(() -> createImageData(jarFile, entry), jarFile.getName(), entry.getName()));
+					ImageElement imageElement = new ImageElement(() -> createImageData(jarFile, entry),
+							jarFile.getName(), entry.getName());
+					mElementsCache.put(imageElement.getPath(), imageElement);
```

Comment:
Can't this code be left unchanged?
```suggestion
					addImageElement(new ImageElement(() -> createImageData(jarFile, entry), jarFile.getName(), entry.getName()));
```